src/server/src/proxy.ts

```diff
@@ -640,74 +605,76 @@ app.post(settings.request_path, (req: Request, res: Response) => {
   processRequest(req.body, req, res)
 })
 
-async function processRequest(query: any, req: Request, res: Response) {
-  if (query.action !== 'tokenorder_check') {
-    logThis('RPC request received from ' + req.ip + ': ' + query.action, log_levels.info)
-    rpcCount++
-  }
-
-  if (settings.use_tokens) {
+async function processTokensRequest(query: NanoRPCRequest, req: Request, res: Response<TokenAPIResponses>): Promise<Response> {
+  switch (query.action) {
     // Initiate token purchase
-    var token_key = ""
-    if (query.action === 'tokens_buy') {
-      var token_amount = 0
-      if ('token_amount' in query) {
+    case 'tokens_buy':
+      let token_key = ""
+      let token_amount = 0
+      if (query.token_amount) {
         token_amount = Math.round(query.token_amount)
       }
       else {
         return res.status(500).json({ error: 'The amount of tokens (token_amount) to purchase must be provided'})
       }
-      if ('token_key' in query) {
+      if (query.token_key) {
         token_key = query.token_key
       }
 
       let payment_request = await Tokens.requestTokenPayment(token_amount, token_key, order_db, settings.node_url)
-
-      res.json(payment_request)
-      return
-    }
+      return res.json(payment_request)
 
     // Verify order status
-    if (query.action === 'tokenorder_check') {
-      if ('token_key' in query) {
-        token_key = query.token_key
+    case 'tokenorder_check':
+      if (query.token_key) {
+        let token_key = query.token_key
         let status = await Tokens.checkOrder(token_key, order_db)
         return res.json(status)
       }
       else {
         return res.status(500).json({ error: 'No token key provided'})
       }
-    }
 
     // Claim back private key and replace the account
-    if (query.action === 'tokenorder_cancel') {
-      if ('token_key' in query) {
-        token_key = query.token_key
+    case 'tokenorder_cancel':
+      if (query.token_key) {
+        let token_key = query.token_key
         let status = await Tokens.cancelOrder(token_key, order_db)
         return res.json(status)
       }
       else {
         return res.status(500).json({ error: 'No token key provided'})
       }
-    }
 
     // Verify order status
-    if (query.action === 'tokens_check') {
+    case 'tokens_check':
       if ('token_key' in query) {
-        token_key = query.token_key
+        let token_key = query.token_key
         let status = await Tokens.checkTokens(token_key, order_db)
         return res.json(status)
       }
       else {
         return res.status(500).json({ error: 'No token key provided'})
       }
-    }
+
+    // Check token price
+    case 'tokenprice_check':
+      let status = await Tokens.checkTokenPrice()
+      return res.json(appendRateLimiterStatus(res, status))
+
+    default:
+      return res.status(404)
```

Comment:
This should perhaps be something else? 500 to be in line with reste of the API?

Reply:
Do you mean the res.status(404)? In that case, maybe
res.status(500).json({ error: 'Invalid token API request'})

Reply:
Yup. This is not something that should happen. Added a logThis statement as well, and changed to HTTP 500.